### DISCUSSION TITLE
Limitations on description field

### DISCUSSION THREAD

**Discussion by alerque** [rust-lang/crates.io#8739] [Q&A] [answered]
I'm one of the new maintainers on the Project Fluent crate and its related co-horts (`fluent`, `fluent-bundle`, `fluent-fallback`, etc.). One of the current shortcomings is a really confusing set of crates spread across two workspaces with documentation that conflates them all over the place. Some of the current crate descriptions don't really make it clear what the purpose of the crate is or when to use which one.

I'm currently working on [overhauling the README files](https://github.com/projectfluent/fluent-rs/pull/359) to try to make it clearer what each crate is for and keep the generic stuff about the ecosystem in general to a minimum.

In the process I'm trying to match thet `description` field so that each crate has a summary that not only introduces the Fluent localization interface in general but identifies the purpose of the crate relative to the project.

My question is how long the description field is allowed to be on crates.io, both in terms of max-length but also more realistically what length it might be truncated to in crates lists, searches, etc. I'm trying to optimize both identifying the crate purpose and what Fluent is all about, but I don't know what the parameters are.

I couldn't find any documentation for the `description` field should be besides being plain text and not Markdown.

**Comment by Turbo87** [comment, marked as answer]
Thank you for asking this question, which made me realize that apparently we don't have a limit... 🙈 

The crate with the longest description is https://crates.io/crates/docker-rust-api at 2503 characters, though it doesn't looks like that is a legitimate use of the description field. On the frontend we truncate the description to 200 characters in e.g. the search results.

**Reply by alerque** [reply]
Thanks for the response.

Presumably since the manifest doesn't specify not having an maximum is fine, but also truncating the UI seems reasonable. Something south of 2500 characters would make sense since the right place for long form content is the README anyway.

Also worth noting it looks like `cargo search` truncates at 98 characters.

**Reply by Turbo87** [reply]
https://github.com/rust-lang/crates.io/pull/8746 has introduced a length limit of 1000 characters now
